src/sentry/similarity/features.py

```diff
@@ -172,40 +178,49 @@ def classify(self, events):
                     )
                 else:
                     if features:
-                        items.append((self.aliases[label], features, ))
-        return zip(
-            map(
-                lambda (alias, characteristics): self.aliases.get_key(alias),
-                items,
+                        items.append((self.aliases[label], thresholds.pop(label, 0), features))
+                        labels.append(label)
+
+        if thresholds:
+            raise ValueError('Invalid thresholds provided')
```

Comment:
This also feels kludgy also

Reply:
This could do with either a comment or a more explicit `if len(thresholds) != 0`

Reply:
This was super broken, so I removed it for now in 7e9905a7d4fb2d4cc6aa7f9aaece1fe76fcf1f41.

The intention here was to ensure all thresholds provided were actually defined in the feature set, but the `pop` actually occurred in a loop here so the threshold would only apply to the first iteration. If we want to validate this in the future, it should probably just be on a set difference between `set(self.features)` and `set(thresholds)` without the destructing.